aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -546,29 +546,29 @@ Tensor nuclear_norm(const Tensor& self, bool keepdim) {
       self.dim() == 2,
       "Expected a tensor with 2 dimensions, but got a tensor with ",
       self.dim(), " dimension", self.dim()==1 ? "" : "s", " instead.");
-  return at::sum(std::get<1>(at::svd(self)), 0, keepdim);
+  return at::sum(std::get<1>(at::svd(self, /*some=*/true, /*compute_uv=*/self.requires_grad())), 0, keepdim);
 }
 
 Tensor &nuclear_norm_out(Tensor& result, const Tensor& self, bool keepdim) {
   TORCH_CHECK(
       self.dim() == 2,
       "Expected a tensor with 2 dimensions, but got a tensor with ",
       self.dim(), " dimension", self.dim()==1 ? "" : "s", " instead.");
-  return at::sum_out(result, std::get<1>(at::svd(self)), 0, keepdim);
+  return at::sum_out(result, std::get<1>(at::svd(self, /*some=*/true, /*compute_uv=*/self.requires_grad())), 0, keepdim);
 }
 
 Tensor nuclear_norm(const Tensor& self, IntArrayRef dim, bool keepdim) {
   TORCH_CHECK(dim.size() == 2, "nuclear norm requires a 'dim' argument of size 2");
 
   Tensor p = _move_to_end(self, dim);
-  return at::sum(std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/false)), -1, keepdim);
+  return at::sum(std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/self.requires_grad())), -1, keepdim);
 }
 
 Tensor& nuclear_norm_out(Tensor& result, const Tensor& self, IntArrayRef dim, bool keepdim) {
   TORCH_CHECK(dim.size() == 2, "nuclear norm requires a 'dim' argument of size 2");
 
   Tensor p = _move_to_end(self, dim);
-  return at::sum_out(result, std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/false)), -1, keepdim);
+  return at::sum_out(result, std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/self.requires_grad())), -1, keepdim);
```

Comment:
does this respect `GradMode` yet?

Reply:
I'm not sure I follow, are you asking if this fix works?

Reply:
Sorry for not being clear. I am asking if `.requires_grad` respects `GradMode` now (e.g., set by `no_grad()`). @yf225 probably knows..

Reply:
No worries, thanks for the clarification. I’m not sure about it either.

Reply:
`.requires_grad` doesn't respect `GradMode` (and probably won't based on the concerns at https://github.com/pytorch/pytorch/pull/22506#pullrequestreview-259641452). Another concern is that `self` might not be a Variable when the user calls this function, and we should likely use `GradMode::is_enabled() && self.is_variable() && self.requires_grad()` here to be safe.

Reply:
Thank you @yf225, I’ll change as advised.

Reply:
@yf225 I've made the change, could you please review when you get the time?